app/modules/site_settings/resources.py

```diff
@@ -111,91 +114,80 @@ def post(self, args):
         site_setting = SiteSetting.set(**args)
 
-        if args.get('key') == 'social_group_roles':
-            SocialGroup.site_settings_updated()
         return site_setting
 
 
-@api.route('/<string:site_setting_key>')
+@api.route('/file/<string:file_key>')
 @api.response(
     code=HTTPStatus.NOT_FOUND,
-    description='SiteSetting not found.',
+    description='File not found.',
 )
-@api.resolve_object_by_model(SiteSetting, 'site_setting', 'site_setting_key')
+@api.resolve_object_by_model(SiteSetting, 'file', 'file_key')
 class SiteSettingFileByKey(Resource):
     """
-    Manipulations with a specific SiteSetting.
+    Manipulations with a specific File.
     """
 
     @api.permission_required(
         permissions.ObjectAccessPermission,
         kwargs_on_request=lambda kwargs: {
-            'obj': kwargs['site_setting'],
+            'obj': kwargs['file'],
             'action': AccessOperation.READ,
         },
     )
-    def get(self, site_setting):
+    def get(self, file):
         """
-        Get SiteSetting details by ID.
+        Get File details by ID.
         """
-        if site_setting.file_upload_guid:
+        if file.file_upload_guid:
             return redirect(
                 url_for(
                     'api.fileuploads_file_upload_src_u_by_id_2',
-                    fileupload_guid=site_setting.file_upload_guid,
+                    fileupload_guid=file.file_upload_guid,
                 )
             )
+        else:
+            abort(400, 'File endpoint only for manipulation of files')
+
         schema = schemas.DetailedSiteSettingFileSchema()
-        json_msg, err = schema.dump(site_setting)
+        json_msg, err = schema.dump(file)
         return json_msg
 
     @api.permission_required(
         permissions.ObjectAccessPermission,
         kwargs_on_request=lambda kwargs: {
-            'obj': kwargs['site_setting'],
+            'obj': kwargs['file'],
             'action': AccessOperation.DELETE,
         },
     )
     @api.login_required(oauth_scopes=['site-settings:write'])
     @api.response(code=HTTPStatus.CONFLICT)
     @api.response(code=HTTPStatus.NO_CONTENT)
-    def delete(self, site_setting):
+    def delete(self, file):
         """
-        Delete a SiteSetting by ID.
+        Delete a File by ID.
         """
-        setting_key = site_setting.key
         context = api.commit_or_abort(
             db.session,
-            default_error_message=f'Failed to delete the SiteSetting "{site_setting.key}".',
+            default_error_message=f'Failed to delete the SiteSetting "{file.key}".',
         )
         with context:
-            db.session.delete(site_setting)
+            db.session.delete(file)
 
-        if setting_key == 'social_group_roles':
-            from app.modules.social_groups.models import SocialGroup
-
-            SocialGroup.site_settings_updated()
         return None
 
 
-# @configuration.route('/definition/main/<path:path>')
+# TODO remove DEX 675
```

Comment:
Temporary to allow the frontend to use both APIs as this is necessary to pass integ tests